tools/boilerplate.py

```diff
@@ -137,155 +140,104 @@ def boilerplate_gen():
-            # be retired soon.
-            if not varargs and fmt is PLOT_TEMPLATE and has_data:
-                args.pop()
-                defaults = defaults[:-1]
-
-            if varargs is not None:
-                call.append('*' + varargs)
-            if varkw is not None:
-                call.append('**' + varkw)
-            call = ', '.join(call)
-
-            text_wrapper.width = 80 - 19 - len(func)
-            join_with = '\n' + ' ' * (18 + len(func))
-            call = join_with.join(text_wrapper.wrap(call))
-
-            if not varargs and fmt is PLOT_TEMPLATE and has_data:
-                args.append('data')
-                defaults = defaults + (None,)
-
-            # Now we can build the argspec for defining the wrapper
-            argspec = inspect.formatargspec(args, varargs, varkw, defaults,
-                                            formatvalue=format_value)
-            argspec = argspec[1:-1]  # remove parens
-
-            text_wrapper.width = 80 - 5 - len(func)
-            join_with = '\n' + ' ' * (5 + len(func))
-            argspec = join_with.join(text_wrapper.wrap(argspec))
-
-            # A gensym-like facility in case some function takes an
-            # argument named washold, ax, or ret
-            washold, ret, ax = 'washold', 'ret', 'ax'
-            bad = {*args, varargs, varkw}
-            while washold in bad or ret in bad or ax in bad:
-                washold = 'washold' + str(random.randrange(10 ** 12))
-                ret = 'ret' + str(random.randrange(10 ** 12))
-                ax = 'ax' + str(random.randrange(10 ** 12))
-
-            # Since we can't avoid using some function names,
-            # bail out if they are used as argument names
-            for reserved in ('gca', 'gci'):
-                if reserved in bad:
-                    msg = 'Axes method %s has kwarg named %s' % (func, reserved)
-                    raise ValueError(msg)
-
-            yield fmt % locals()
+                self._repr = repr(value)
+
+        def __repr__(self):
+            return self._repr
+
+    text_wrapper = textwrap.TextWrapper(
+        break_long_words=False, width=70,
+        initial_indent=' ' * 8, subsequent_indent=' ' * 8)
+
+    for func in _commands:
+        # For some commands, an additional line is needed to set the color map.
+        if func in cmappable:
+            fmt = CMAPPABLE_TEMPLATE
+            mappable = '    ' + cmappable[func]
+        else:
+            fmt = NON_CMAPPABLE_TEMPLATE
+
+        # Get signature of wrapped function.
+        sig = inspect.signature(getattr(Axes, func))
+
+        # Replace self argument.
+        params = list(sig.parameters.values())[1:]
+
+        sig = str(sig.replace(parameters=[
+            param.replace(default=value_formatter(param.default))
+            if param.default is not param.empty else param
+            for param in params]))
+        # Move opening parenthesis before newline.
```

Comment:
Only wrap for longer signatures by making this conditional:

`if 4 + len(func) + len(sig) >= 80:`.

Reply:
feel free to push the change to my branch

Reply:
Honestly I think that when the source looks like
```
In [1]: plt.errorbar??
Signature: plt.errorbar(x, y, yerr=None, xerr=None, fmt='', ecolor=None, elinewidth=None, capsize=None, barsabove=False, lolims=False, uplims=False, xlolims=False, xuplims=False, errorevery=1, capthick=None, *, data=None, **kwargs)
Source:   
@docstring.copy_dedent(Axes.errorbar)
def errorbar(
        x, y, yerr=None, xerr=None, fmt='', ecolor=None,
        elinewidth=None, capsize=None, barsabove=False, lolims=False,
        uplims=False, xlolims=False, xuplims=False, errorevery=1,
        capthick=None, *, data=None, **kwargs):
    return gca().errorbar(
        x=x, y=y, yerr=yerr, xerr=xerr, fmt=fmt, ecolor=ecolor,
        elinewidth=elinewidth, capsize=capsize, barsabove=barsabove,
        lolims=lolims, uplims=uplims, xlolims=xlolims,
        xuplims=xuplims, errorevery=errorevery, capthick=capthick,
        data=data, **kwargs)
File:      ~/src/extern/matplotlib/lib/matplotlib/pyplot.py
Type:      function
```
the biggest issue is not with an extra newline, but that that's a useless source...

We could set `__wrapper__` to the inner function so that getsource points to `Axes.errorbar`, but that may be confusing for functions that do call sci() with the return value...